### PR TITLE
[TSK-140] add user-manager CI

### PR DESCRIPTION
**Pull request by kostya4201337**

### CODE REVIEW

**Comment by konstantinwas** [comment]
well it doesn't work, i'll fix it later

File: user-manager/.github/workflows/gradle.yaml
Comment (adsukharev) [inline user-manager/.github/workflows/gradle.yaml:1]: rename file to ci.yaml

**Review by adsukharev — COMMENTED** [review]
add a simple test to validate that it works, kind of like this:
```
void test() {
   assertThat(true).isTrue();
}
```
if it works than break it to check failure:
`assertThat(false).isTrue(); `

then remove the test before merging

File: user-manager/.github/workflows/gradle.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@bd5760595778326ba7f1441bcf7e88b49de61a25
+        with:
+          arguments: build
```
Comment (adsukharev) [inline user-manager/.github/workflows/gradle.yaml]: you have to move your ci pipeline from user-manager folder to the root. Then it should work

**Comment by konstantinwas** [comment]
👍

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@bd5760595778326ba7f1441bcf7e88b49de61a25
+        with:
+          arguments: build
+      - name: Test with Gradle
+        run: gradle clean test
```
Comment (adsukharev) [inline .github/workflows/ci.yaml]: It's a good idea to separate building and testing the project.
`gradle build` operation compiles and tests the project. Then you run `gradle test` which tests it again. It's a commonly used operations locally but if we want to separate steps it does not suit us. 
You need to compile in one step, in a second test it. Try it out:
```
- name: Build 
 run: gradle classes
      
- name: Test
run: gradle test
```

File: .gitignore [hunk 3]
```diff
@@ -24,7 +24,7 @@ bin/
 *.ipr
 out/
 !**/src/main/**/out/
-!**/src/test/**/out/
+!**/src/Test/**/out/
```
Comment (adsukharev) [inline .gitignore]: `test` folder should be in lowecase

File: .gitignore [hunk 4]
```diff
@@ -3,7 +3,7 @@ user-manager/HELP.md
 build/
 !gradle/wrapper/gradle-wrapper.jar
 !**/src/main/**/build/
-!**/src/test/**/build/
+!**/src/Test/**/build/
```
Comment (adsukharev) [inline .gitignore:6]: Lowercase

File: .gitignore [hunk 5]
```diff
@@ -15,7 +15,7 @@ build/
 .sts4-cache
 bin/
 !**/src/main/**/bin/
-!**/src/test/**/bin/
+!**/src/Test/**/bin/
```
Comment (adsukharev) [inline .gitignore:18]: Lowercase

**Review by adsukharev — APPROVED** [review]

**Review by Hoocki — APPROVED** [review]

**Pull request merged**